docs/tutorials/mongoid-queries.txt

```diff
@@ -943,11 +943,19 @@ Query Cache
 -----------
 
 If Mongoid's QueryCache is enabled, it will cache queries that the server returns in a single batch
-and avoids sending requests to the database for identical queries. Note that if the number of
-results is too large to be returned in a single batch, the QueryCache will *not* be used, even if
-Mongoid::QueryCache.enabled is true.
+and avoids sending requests to the database for identical queries. Queries are cached once per each thread.
+Note that if the number of results is too large to be returned in a single batch,
+the QueryCache will *not* be used, even if Mongoid::QueryCache.enabled is true.
 
-To enable the QueryCache:
+When using Rails or another Rack-based framework, Mongoid provides middleware
+which clears the QueryCache after each request. To enable the QueryCache via middleware (recommended):
+
+.. code-block:: ruby
+
+  # in an initializer
+  Rails.application.config.middleware.use Mongoid::QueryCache::Middleware
```

Comment:
Could you change this to be aligned with the documentation in https://guides.rubyonrails.org/rails_on_rack.html#adding-a-middleware as to how to add middleware?

Reply:
Done